OutOfSchool/OutOfSchool.WebApi.Tests/Services/ProviderServiceTests.cs

```diff
@@ -1064,6 +1064,49 @@ public async Task Block_ReturnsNull_IfNotAdmin()
 
     #endregion Block
 
+    #region ValidateImportData
+
+    [Test]
+    public void ValidateImportData_ReturnsArgumentNullException_IfDtoIsNull()
+    {
+        // Arrange
+
+        // Act, Assert
+        Assert.ThrowsAsync<ArgumentNullException>(async () => await providerService.ValidateImportData(null).ConfigureAwait(false));
+    }
+
+    [Test]
+    public async Task ValidateImportData_DontExecuteGettingData_IfDtoIsEmpty()
+    {
+        // Arrange
+        var data = new ImportDataValidate();
+
+        // Act
+        var result = await providerService.ValidateImportData(data).ConfigureAwait(false);
+
+        // Assert
+        providersRepositoryMock.Verify(x => x.CheckExistsByEdrpous(data.Edrpous), Times.Never);
+        providersRepositoryMock.Verify(x => x.CheckExistsByEmails(data.Emails), Times.Never);
+    }
+
+    [Test]
+    public async Task ValidateImportData_ExecuteGettingData_IfDtoIsValid()
+    {
+        // Arrange
+        var data = new ImportDataValidate();
+        data.Edrpous.Add("test");
+        data.Emails.Add("test");
+
+        // Act
+        var result = await providerService.ValidateImportData(data).ConfigureAwait(false);
+
+        // Assert
+        providersRepositoryMock.Verify(x => x.CheckExistsByEdrpous(data.Edrpous), Times.Once);
+        providersRepositoryMock.Verify(x => x.CheckExistsByEmails(data.Emails), Times.Once);
+    }
```

Comment:
(Test atomicity) Please consider adding to this test 2 more. 

In your current implementation of `ValidateImportData()` the `CheckExistsByEdrpous()` and `CheckExistsByEmails()` are called independently. So there should be independent tests as well. 

One existing - when both fields are set.

One with only `Edrpous` set, with assertion like
```
providersRepositoryMock.Verify(x => x.CheckExistsByEdrpous(data.Edrpous), Times.Once);
providersRepositoryMock.Verify(x => x.CheckExistsByEmails(It.IsAny<List<string>>))), Times.Never);
```

And with only `emails` set, with assertion like
```
providersRepositoryMock.Verify(x => x.CheckExistsByEdrpous(It.IsAny<List<string>>()), Times.Never);
providersRepositoryMock.Verify(x => x.CheckExistsByEmails(data.Emails), Times.Once);
```

Reply:
I did it